src/uucore/src/lib/mods/error.rs

```diff
@@ -0,0 +1,198 @@
+    }
+}
+
+impl Display for UError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            UError::Common(e) => e.fmt(f),
+            UError::Custom(e) => e.fmt(f),
+        }
+    }
+}
+
+/// All errors should implement `std::error::Error` and have an additional
+/// code` method that specifies the exit code of the program if the error
+/// is returned from `uumain`.
+pub trait UCustomError: Error {
+    fn code(&self) -> i32;
+}
+
+impl Into<i32> for Box<dyn UCustomError> {
+    fn into(self) -> i32 {
+        self.code()
+    }
+}
+
+/// A common error with an overriden exit code. This exit code is returned
+/// instead of the default exit code for the `CommonError`
+#[derive(Debug)]
+pub struct UCommonErrorWithCode {
+    code: i32,
+    error: UCommonError,
+}
+
+impl Error for UCommonErrorWithCode {}
+
+impl Display for UCommonErrorWithCode {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
+        self.error.fmt(f)
+    }
+}
+
+impl UCustomError for UCommonErrorWithCode {
+    fn code(&self) -> i32 {
+        self.code
+    }
+}
+
+#[derive(Debug)]
+pub struct UErrorWithCodeAndMessage {
+    code: i32,
+    message: String,
+}
+
+impl Error for UErrorWithCodeAndMessage {}
+
+impl Display for UErrorWithCodeAndMessage {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
+        self.message.fmt(f)
+    }
+}
+
+impl UCustomError for UErrorWithCodeAndMessage {
+    fn code(&self) -> i32 {
+        self.code
+    }
+}
+
+/// Common errors for utilities. If identical errors appear across multiple
+/// utilities, they should be added here.
+#[derive(Debug)]
+pub enum UCommonError {
+    NoSuchFileOrDirectory(PathBuf),
+}
+
+impl UCommonError {
+    pub fn with_code(self, code: i32) -> UCommonErrorWithCode {
+        UCommonErrorWithCode { code, error: self }
+    }
+
+    pub fn code(&self) -> i32 {
+        match self {
+            &UCommonError::NoSuchFileOrDirectory(_) => 1,
+        }
+    }
+}
+
+impl Into<i32> for UCommonError {
+    fn into(self) -> i32 {
+        self.code()
+    }
+}
+
+impl Error for UCommonError {}
+
+impl Display for UCommonError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
+        match self {
+            UCommonError::NoSuchFileOrDirectory(p) => {
+                write!(
+                    f,
+                    "cannot access '{}': No such file or directory",
+                    p.display()
+                )
+            }
+        }
+    }
+}
+
+/// A special error type that does not print any message when returned from
+/// `uumain`.
+#[derive(Debug)]
+pub struct ExitCode(i32);
+
+impl ExitCode {
+    pub fn new(code: i32) -> Box<ExitCode> {
+        Box::new(ExitCode(code))
+    }
+}
+
+impl Error for ExitCode {}
+
+impl Display for ExitCode {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
+        write!(f, "")
```

Comment:
I think we can just return Ok instead.
```suggestion
        Ok(())
```

Reply:
Ah, of course! Thanks!